pkgs/stdenv/linux/default.nix

```diff
@@ -115,8 +115,6 @@ let
     ''
       export NIX_ENFORCE_PURITY="''${NIX_ENFORCE_PURITY-1}"
       export NIX_ENFORCE_NO_NATIVE="''${NIX_ENFORCE_NO_NATIVE-1}"
-      ${lib.optionalString (system == "x86_64-linux") "NIX_LIB64_IN_SELF_RPATH=1"}
-      ${lib.optionalString (system == "mipsel-linux") "NIX_LIB32_IN_SELF_RPATH=1"}
```

Comment:
I'm afraid it's not that easy and depends on specific ports.

In theory non-multilib targets should just use `lib`. In practice what is a multilib has a moot definition: for example `x86_64-linux-musl` is always multilib while `x86_64-linux-musl` is (or, should be) never multilib.

You can see remnants of multilib in `gcc` itself:

```
$ gcc -print-multi-os-directory
../lib64
```

Sometimes packages are diligent enough to have `lib -> lib64` symlink (or the other way around). Sometimes they are not.

The question here is: What API do we expect for `nixpkgs` to provide and packages to use for non-multilib? Is it always `lib` or something else (`lib64`)? Today we do both with `lib64` being a canonical path.

This change effectively says `lib64` is not considered anymore as a searchpath.

Reply:
Cross-compilation case is a great point! I'll spend some time today to understand how `nixpkgs` plumbs `rpath` details there and get back.

Reply:
Is it fair to say `NIX_LIB*_IN_SELF_RPATH` should only be set for `multiStdenv`? If so, the variables could be set conditionally there.

Reply:
> Is it fair to say NIX_LIB*_IN_SELF_RPATH should only be set for multiStdenv? If so, the variables could be set conditionally there.

Ideally (if non-multilib environment used `lib` everywhere consistently) - yes. In practice I don't think we are there yet. We probably want both `lib` and `lib64` to still be pulled in. Normally I would expect `NIX_LDFLAGS` to apply to host toolchains (and `NIX_LDFLAGS_FOR_BUILD` / `NIX_LDFLAGS_FOR_TARGET` be disambiguating variables).

On the other hand looking at the history far back it was added before cross-compilation was a well-supported target: https://github.com/NixOS/nixpkgs/commit/d9213df2c107753963f99e34a100c8e16ed76d06 . It probably outlived it's purpose and we have many other ways to influence the driver.

Let's try your change as is and see if it works on multilib.

Reply:
> Let's try your change as is and see if it works on multilib.

Nice, thank you.

> It probably outlived it's purpose and we have many other ways to influence the driver.

By "other means to influence the driver", do you mean the corresponding (and only) users of the `NIX_LIB*_IN_SELF_RPATH`,

https://github.com/NixOS/nixpkgs/blob/2ba1f8684d452afeef3ddc5e1f4cf7cda68646fc/pkgs/stdenv/generic/setup.sh#L314-L319

should be removed as well?

Reply:
Yeah, let's remove those as well.

Reply:
Done.